### ISSUE TITLE
Missing mutations for ListView items

### ISSUE THREAD

**Issue by etvorun** [dotnet/maui#9640]
### Description

XAML Hot Reload does not work when editing DataTemplate used by ListView

### Steps to Reproduce

1. Open attached project
2. Build and run
3. In VS open Debug > Windows > Live Visual Tree
4. Toggle off Just My XAML button in LVT toolbar
5. Expand elements
**Expected**: there are elements under ListView
**Actual**: no elements under ListView
**Note**: as a result XAML Hot Reload does not work when editing content of ListView's DataTemplate 
[MauiApp1.zip](https://github.com/dotnet/maui/files/9419324/MauiApp1.zip)

![image](https://user-images.githubusercontent.com/48451158/186521441-21df80e9-6f37-4f75-9065-6b797fad4e13.png)

 

### Version with bug

6.0.408

### Last version that worked well

Unknown/Other

### Affected platforms

Windows, I was *not* able test on other platforms

### Affected platform versions

Windows

**Comment by drasticactions** [comment]
I believe the root of this problem is how ListView handles child elements, which is the same as Xamarin.Forms https://github.com/xamarin/Xamarin.Forms/issues/14277.

When XAML Hot Reload checks for child elements, we use 'IVisualTreeElement' and get the visual children from the parent. This is implemented on the base element, but ListView doesn't store its children there. It has its own internal lists.

@etvorun had pointed out [ItemsView](https://github.com/dotnet/maui/blob/92e876e4842443b50b5bf0d6a7edbea46de2e646/src/Controls/src/Core/Items/ItemsView.cs) and that this could be the place to implement that interface.

**Comment by drasticactions** [comment]
@jfversluis I do not believe this is a Visual Studio bug, but a MAUI one. https://github.com/xamarin/Xamarin.Forms/issues/14277

This can only be fixed if IVisualTreeElement is implemented for ItemsView or ListView. If it's not, XAML Hot Reload can't work for those controls.

**Comment by drasticactions** [comment]
I am going to spike what it would take to fix this. It is a significant enough control that it's worth exploring it.

**Comment by Expecho** [comment]
@jfversluis @homeyf  Any update on this?

**Comment by jfversluis** [comment]
We're looking at deprecating ListView for .NET 10, so I don't think there will be a lot of priority for this unfortunately. Please use CollectionView

**Comment by drasticactions** [comment]
If you invoke [Full Page XAML Hot Reload](https://learn.microsoft.com/en-us/dotnet/maui/xaml/hot-reload?view=net-maui-9.0#full-page-reload) (By invoking the Hot Reload button or setting it up to be invoked when you save the current page) then it should hot reload the entire page, causing the list view to update (Although it will lose your current placement within the page to do this, since it needs to reload the full page)

With ListView itself being deprecated, this is likely not to be addressed. It only affects incremental Hot Reload, and while that's bad, there are workarounds for it, and potential fixes for it are _very_ disruptive and likely to break ListView cells in other areas, which you don't want.

**Comment by jfversluis** [comment]
ListView is deprecated for .NET 10, so we probably won't be working on this anymore. Please migrate to CollectionView and help us make that the solid approach for this. Thanks!
